V1/Services/WalletService.cs

```diff
@@ -109,6 +111,24 @@ public async Task<Wallet> CreateWallet(CreateWalletRequest walletRequest, bool m
 
             return wallet;
         }
+
+        public async Task DeleteWallet(string walletId)
+        {
+            var wallet = await _walletRepository.GetAsync(Guid.Parse(walletId));
+
+            if (wallet == null)
+                throw new NotFoundException(Messages.Wallet_NotFound);
+
+            await _walletDetailService.DeleteWalletDetails(wallet);
+
+            await _testResultService.DeleteTestResults(wallet.Id);
+
+            _walletRepository.Delete(wallet);
+            await _walletRepository.SaveAsync();
```

Comment:
NOTE: I always try and return some value, I found it useful if ever i need to reuse something and it has a different application